docs/advanced/ecdsa.md

```diff
@@ -0,0 +1,46 @@
+KICL supports the ECDSA-NIST256P-CHALLENGE SASL authentication mechanism. This document describes both how the mechanism works and how the library can be used to authenticate with this method.
+
+### Protocol description
+
+The ECDSA authentication mechanism makes use of the SASL extension. This must be requested by the client prior to any authentication attempts (`CAP REQ :sasl`). Once the server sends an `ACK`, authentication can begin.
```

Comment:
The mechanism itself doesnt `make use of` the SASL extension. It's more of a layer on top of an extensible system. Thoughts on alternative wording?